src/elements/tweet/tweetDataTransformer.ts

```diff
@@ -0,0 +1,65 @@
+import type { FieldNameToValueMap } from "../../plugin/helpers/fieldView";
+import type { Asset } from "../helpers/defaultTransform";
+import { undefinedDropdownValue } from "../helpers/transform";
+import type { TransformIn, TransformOut } from "../helpers/types/Transform";
+import type { createTweetFields } from "./TweetSpec";
+
+export type ExternalTweetFields = {
+  source: string;
+  isMandatory: string;
+  role: string | undefined;
+  url: string;
+  originalUrl: string;
+  id: string;
+  caption: string;
+  html: string;
+  authorName: string;
+  hideMedia?: string;
+  hideThread?: string;
+};
```

Comment:
Can we use `defaultTransform` for this? It has `transformRole` and `isMandatory` options, and might save us a few lines.